### ISSUE TITLE
cpplint ignored check order of headers

### ISSUE THREAD

**Issue by Mirasire** [cpplint/cpplint#199]
when i include file's own header in the end of all includes, and use cpplint to check the code style.
but cpplint seems ignored the rules of [header file import order](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes) . 
my cpplint version is `1.6.0`, the test code is below.

**`command and output:`**
```bash
$> cpplint --repository=.  test.cpp 
#Done processing test.cpp
```

**`example .cpp file:`**
```cpp
/*
    Copyright 2022 XXX
*/
#include <math.h>
#include <iostream>

#include "./test.h"

int main() {
    std::cout << "hello" << endl;
    return 0;
}
```

**`directory structure:`**
```
.
├── test.cpp
└── test.h
```

**Comment by tkruse** [comment]
Hi, thanks for the report. As far as I can see, cpplint.py has never reported an error for this.
Specifically, cpplint.py does not know if for a file `test.cpp`, the header `test.h` should be the "related header" or Other libraries' .h files. While the example makes it very likely that `test.h` should be first, I am not sure if adding such an error now would break valid use-cases where it should not be first.

So I am inclined to close this as "wontfix" unless upstream repo fixes this.
